app/controllers/portal/portal_controller.rb

```diff
@@ -4,7 +4,14 @@ class PortalController < ApplicationController
     layout "portal"
 
     def home
-      @tax_returns = current_client.tax_returns.order(year: :desc)
+      @current_step = nil
+      @tax_returns = []
+
+      if current_client.completed_intake_process?
+        @tax_returns = current_client.tax_returns.order(year: :desc)
+      else
+        @current_step = current_client.intake.determine_current_step
```

Comment:
I originally had it as current_client.intake.current_step || current_client.intake.determine_current_step, but that seemed kind of redundant since determine_current_step was already returning current_step if it wasn't already determined.

Also, the idea of showing this until they've started prep is by design and was dictated by product/design/customer research. The idea is that we're going to be showing this list of steps up until we've begun preparing their taxes. I spoke to them about it at length and discussed some tradeoffs, but we went this direction at their request.

Reply:
completed_at cannot be used for this value because so many of our existing users _never_ have their intake set as completed -- they're signing their tax returns and their intake was never "officially" marked as completed_at because so many drop off at the document upload step. So we needed to choose some other way to make the cut off between when we stop showing the intake part and when we show the tax return part. Keep in mind that this is an "in between" state to bridge the existing implementation of e-signatures and the introduction of returning client and I don't particuarly expect that this is the way that we will determine this state forever, but its reliable and works for the cases that product/design wants to address in the short term. Upcoming stories will ensure that there are certain features available to users in the "in between" stages (message your tax team, upload a document), but this prevents us from showing "finish your intake" steps to users who we're actually successfully preparing taxes for with the information they've already provided.

Reply:
This also appropriately covers the drop-off scenario because we _don't_ mark those drop-offs as completed_at, but we put them right into the prep state. In the drop-off case, individuals who log in would see the tax return steps with the "there are no documents for you to review yet" message, which seems like what we'd want/expect.